.ci/test-spelling-unknown-words.sh

```diff
@@ -46,7 +46,10 @@ echo "Run w"
   .ci-temp/contribution/jsoref-spellchecker/exclude.pl |\
   xargs -0 $word_splitter |\
   $word_splitter |\
-  perl -p -n -e 's/ \(.*//' > $run_output
+  perl -p -n -e 's/ \(.*//' |\
+  perl -n -e 'next if /^u?[a-f]+$/i; print' |\
+  perl -n -e 'next if /^([a-z])\1+$/i; print' |\
```

Comment:
please explain why this required to be part of script, I think it should be just a change at whitelist file.
In my mind spell checking is specific to a project only by whitelist and excludes for file/dirs(generated, resources, ....).
So as finish enforcing ... it might be good to have all common parts in our repo(s) and we just clone/wget project append to it excludes+whitelist and receive validation.
It will allow you all others easily reuse your spellchecker easily.

Reply:
I tried to clean up content of whitelist and run ci shell file (with disabled clone or source). It suppose to report all words as missed and do not have weird ZZZZZ variables.
But such resources are still processed by spell-checker: 
```
$ ./.ci/test-spelling-unknown-words.sh
...
+ZZZZZZ
EOF

$ grep -r "ZZZZZZ"
.ci-temp/contribution/jsoref-spellchecker/unknown.words:ZZZZZZ
src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocPositionWithSinglelineComments.java:class ZZZZZZ {

```

Reply:
I'm not attached to these filters. I *could* probably teach `w` to optionally support them.
In principle, something produces a lists of files for `w` to process, and some exclusions are applied.
So, two filters, `exclude.pl` to skip files that shouldn't be checked, and this stuff to drop large families of words that I think you don't care about. This could be split into a script. For now, it's inline, just to try.

I tested it here, and it seems to work:
1. Travis generates the expected output: https://api.travis-ci.org/v3/job/366693457/log.txt
2. Applying a lazy change to avoid the reset:
```
@@ -16,1 +16,1 @@ else
-  (cd $contrib; git fetch; git reset --hard origin/master)
+  (cd $contrib; git fetch; echo git reset --hard origin/master)
```
And applying the patch from the output, and then running it again results in no complaints.

I'm not sure what's going wrong.

Reply:
I applied removal of odd names to contribution repo 

I relaunched latest maven build - https://travis-ci.org/checkstyle/checkstyle/jobs/366822545
and It wants them back, so it mean that exclude does not work, and it still validate resoures.

Please review exclude.pl

Reply:
checkstyle/contribution#298 is the fix to the identified exclude.pl bug